step-framework-parent/step-framework-timeseries/src/main/java/step/core/timeseries/TimeSeriesCollection.java

```diff
@@ -0,0 +1,105 @@
+package step.core.timeseries;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import step.core.collections.Collection;
+import step.core.collections.Filter;
+import step.core.collections.IndexField;
+import step.core.timeseries.aggregation.TimeSeriesAggregationPipeline;
+import step.core.timeseries.aggregation.TimeSeriesAggregationQuery;
+import step.core.timeseries.aggregation.TimeSeriesAggregationResponse;
+import step.core.timeseries.bucket.Bucket;
+import step.core.timeseries.bucket.BucketAttributes;
+import step.core.timeseries.bucket.BucketBuilder;
+import step.core.timeseries.ingestion.TimeSeriesIngestionPipeline;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+
+public class TimeSeriesCollection {
+	
+	private static final Logger logger = LoggerFactory.getLogger(TimeSeriesCollection.class);
+	
+	private final Collection<Bucket> collection;
+	private final long resolution;
+	private final TimeSeriesIngestionPipeline ingestionPipeline;
+
+	public TimeSeriesCollection(Collection<Bucket> collection, long resolution) {
+		this(collection, resolution, new TimeSeriesIngestionPipeline(collection, resolution));
+	}
+	
+	public TimeSeriesCollection(Collection<Bucket> collection, long resolution, TimeSeriesIngestionPipeline ingestionPipeline) {
+		this.collection = collection;
+		this.resolution = resolution;
+		this.ingestionPipeline = ingestionPipeline;
+	}
+	
+	public void createIndexes(Set<IndexField> indexFields) {
+        collection.createOrUpdateIndex("begin");
+        Set<IndexField> renamedFieldIndexes = indexFields.stream().map(i -> new IndexField("attributes." + i.fieldName,
+                i.order, i.fieldClass)).collect(Collectors.toSet());
+        renamedFieldIndexes.forEach(collection::createOrUpdateIndex);
+    }
+
+	public Collection<Bucket> getCollection() {
+		return collection;
+	}
+
+	public TimeSeriesIngestionPipeline getIngestionPipeline() {
+		return ingestionPipeline;
+	}
+
+	public long getResolution() {
+		return resolution;
+	}
+	
+	public void ingestPoint(Map<String, Object> attributes, long timestamp, long value) {
+		this.ingestionPipeline.ingestPoint(attributes, timestamp, value);
+	}
+	
+	public void ingestBucket(Bucket bucket) {
+		this.ingestionPipeline.ingestBucket(bucket);
+	}
+	
+	protected TimeSeriesAggregationResponse collect(TimeSeriesAggregationQuery query) {
```

Comment:
I would keep the concept of TimeSeriesAggregationPipeline which completely disapeared here, that would also allow to avoid changes client side (Step)